### PR TITLE
Kanillian C

### PR DESCRIPTION
**Pull request by NatKarmios**

### CODE REVIEW

File: GillianCore/debugging/adapter/inspect.ml [hunk 1]
```diff
@@ -9,13 +9,13 @@ module DL = Debugger_log
 module Make (Debugger : Debugger.S) = struct
   open Custom_commands (Debugger)
 
-  let run dbg rpc =
-    DL.set_rpc_command_handler rpc ~name:"Threads"
+  let run ~dump_dbg dbg rpc =
```
Comment (giltho) [inline GillianCore/debugging/adapter/inspect.ml:12]: If `dump_dbg` is a global thing set once-and-for-all at startup, I'd make it a global variable. It's kinda polluting all signatures right now

Comment (giltho) [reply GillianCore/debugging/adapter/inspect.ml:12]: (Though not sure what its role is, so might no be possible)

File: GillianCore/utils/list_utils.ml [hunk 2]
```diff
@@ -236,3 +236,9 @@ let get_and_remove_nth n l =
         else x :: aux (i + 1) r
   in
   (!found, aux 0 l)
+
+let map_head f = function
+  | [] -> []
+  | x :: xs -> f x :: xs
+
+let map_last f l = l |> List.rev |> map_head f |> List.rev
```
Comment (giltho) [inline GillianCore/utils/list_utils.ml:244]:
```ocaml
let[@tail_mod_cons] map_last f l =
   match l with
   | [] -> []
   | [ a ] -> [ f a ]
   | x::r -> x :: map_last f l
```

**Review by giltho — APPROVED** [review]
mostly approve

File: GillianCore/utils/list_utils.ml (hunk 2 above)
Comment (NatKarmios) [reply GillianCore/utils/list_utils.ml]: Resolved in aaa58c2889789854b009175b6fbd8d3bab648d09

**Pull request merged**